### DISCUSSION TITLE
Converting Posgres field names (snake_case) to JS/TS camelCase field names

### DISCUSSION THREAD

**Discussion by jameshschuler** [supabase/supabase#646] [Questions] [answered]
Anyone know of a way to do this when typing the response from a query?

`
interface SomeType {
    firstName: string;
}
`

`supabase.from<SomeType>( 'person').select( '*' ).eq( 'id', 1 );`

**Comment by soedirgo** [comment, marked as answer]
Hey @jameshschuler!

You can use camelCase for the column names. Otherwise you'll need to convert them manually with something like [camelcase](https://github.com/sindresorhus/camelcase).

**Reply by jameshschuler** [reply]
Okay cool thanks I'll check out camelcase.

**Reply by jameshschuler** [reply]
Definitely would be a nice option to have when creating the client.

**Reply by fergalmoran** [reply]
+1 on this, it would be great to have a "JsonFieldName" sort of property on columns.

**Comment by gerabu** [comment]
You can use column aliases, for instance:
`type Table = { createdAt: Date }`
`const { data } = await supabase.from('table').select('id, name, createdAt:created_at')`
`const [first] = data as Table[]`
`console.log(first.createdAt)`

**Comment by whaaaley** [comment]
Elaborating on the previous suggestion of aliases...

```ts
// Converts a string from snake_case to camelCase
// Example: "hello_world" -> "helloWorld"
const snakeToCamel = (str: string) => {
  return str.replace(/(_\w)/g, (m) => m[1].toUpperCase())
}

// Creates Supabase column aliases that map snake_case DB fields to camelCase properties
// Example: "user_id, first_name" -> "userId:user_id, firstName:first_name"
export const camelAlias = (fields: string) => {
  return fields
    .split(',')
    .map((field) => field.trim())
    .map((snakeCase) => snakeToCamel(snakeCase) + ':' + snakeCase)
    .join(',')
}
```

```ts
// Usage:
const { data: parties, error } = await client
  .from('parties')
  .select(camelAlias('id, user_id, name, slot_1, slot_2, slot_3, slot_4, slot_5, slot_6, created_at'))
```

**Reply by whaaaley** [reply]
This actually prevents Supabase client from typing the resulting data correctly for some reason...
